python/ray/experimental/channel/nccl_group.py

```diff
@@ -188,22 +205,38 @@ def recv(
             raise RayChannelError("NCCL group has been destroyed.")
         assert allocator is not None, "NCCL group requires a tensor allocator"
         buf = allocator(shape, dtype)
-        self._comm.recv(
-            self.nccl_util.get_tensor_ptr(buf),
-            buf.numel(),
-            self.nccl_util.get_nccl_tensor_dtype(buf),
-            peer_rank,
-            self._cuda_stream.ptr,
-        )
+        import cupy as cp
+
+        if self._dedicated_streams:
+            event = cp.cuda.Event()
+            self._comm.recv(
+                self.nccl_util.get_tensor_ptr(buf),
+                buf.numel(),
+                self.nccl_util.get_nccl_tensor_dtype(buf),
+                peer_rank,
+                self._recv_stream.ptr,
+            )
+            event.record(self._recv_stream)
```

Comment:
actually I can imagine we can also do recv_stream.synchronize()

what's the pros of cons of using event vs stream.synchornize?

Reply:
hmm, at the consumer side (compute operation), we'd like to sync on a particular event on the recv_stream, rather than the whole stream (there might be other operations launched to the same recv_stream).